### PR TITLE
GSdx: Adjust ShinOnimusha crc hack.

### PR DESCRIPTION
**Pull request by ivan89el**
I do not see any problems with graphics. Let's move the operating hack to the aggressive mode, because it increases the performance (some smoke and fog effects are turned off).

### CODE REVIEW

**Comment by lightningterror** [comment]
Can you be more precise what each hack does ?

Also you might wanna check 
https://github.com/PCSX2/pcsx2/issues/642
https://github.com/PCSX2/pcsx2/issues/1481

**Comment by ivan89el** [comment]
Do I have to go through the whole game?
In the future I will test onimsha warlords.

**Comment by lightningterror** [comment]
Well we don't know what the hacks do, it's better to test multiple levels.

**Comment by ivan89el** [comment]
I went through several levels and did not notice any problems. But my computer is bad for the game and I can not go through with it. It is worth writing "needs testing".

**Comment by ivan89el** [comment]
Can be combined. During the passage, there was not one graphic problem that this hack would correct.

**Comment by lightningterror** [comment]
What about vram usage/spikes or slowdowns ?

**Comment by ivan89el** [comment]
With memory everything is normal: the leak refers to 1 (warlords) and 2 (Samurai's Destiny) parts. In some places (a lot of smoke and fog), the hack improves performance on weak systems (so it was moved to aggressive)

File: plugins/GSdx/GSHwHack.cpp [hunk 1]
```diff
@@ -2040,6 +2009,37 @@ bool GSC_SimpsonsGame(const GSFrameInfo& fi, int& skip)
 	return true;
 }
 
+bool GSC_ShinOnimusha(const GSFrameInfo& fi, int& skip)
+{
+	if(Aggressive && skip == 0)
+	{
+
```
Comment (lightningterror) [inline plugins/GSdx/GSHwHack.cpp]: You can remove this extra line here.

File: plugins/GSdx/GSHwHack.cpp [hunk 2]
```diff
@@ -2040,6 +2009,37 @@ bool GSC_SimpsonsGame(const GSFrameInfo& fi, int& skip)
...
+		}
+		else if(fi.TPSM == PSM_PSMT8H && fi.FBMSK == 0xFF000000)
+		{
+			skip = 1;
+		}
+		else if(fi.TME && (fi.TBP0 ==0x1400 || fi.TBP0 ==0x1000 ||fi.TBP0 == 0x1200) && (fi.TPSM == PSM_PSMCT32 || fi.TPSM == PSM_PSMCT24))
+		{
+			skip = 1;
+		}
+
```
Comment (lightningterror) [inline plugins/GSdx/GSHwHack.cpp]: Ditto.

File: plugins/GSdx/GSHwHack.cpp [hunk 3]
```diff
@@ -2265,6 +2264,7 @@ void GSState::SetupCrcHack()
 		lut[CRC::SMTDDS2] = GSC_SMTNocturneDDS<0x20435BF0>;
 		lut[CRC::SMTNocturne] = GSC_SMTNocturneDDS<0x2054E870>;
 		lut[CRC::SoTC] = GSC_SoTC;
+		lut[CRC::ShinOnimusha] = GSC_ShinOnimusha;
```
Comment (lightningterror) [inline plugins/GSdx/GSHwHack.cpp]: This should be below GSC_ResidentEvil4.

File: plugins/GSdx/GSHwHack.cpp [hunk 4]
```diff
@@ -2040,6 +2009,37 @@ bool GSC_SimpsonsGame(const GSFrameInfo& fi, int& skip)
...
+		else if(fi.TME && (fi.TBP0 ==0x1400 || fi.TBP0 ==0x1000 ||fi.TBP0 == 0x1200) && (fi.TPSM == PSM_PSMCT32 || fi.TPSM == PSM_PSMCT24))
+		{
+			skip = 1;
+		}
+
+	}
+
+	return true;
+}
+
```
Comment (lightningterror) [inline plugins/GSdx/GSHwHack.cpp]: Likewise with the entire hack, should be moved below the ResidentEvil4.
Keeping it tidy in alphabetical order.

**Comment by lightningterror** [comment]
@ivan89el can you update the PR so I cam merge it ?

**Comment by ivan89el** [comment]
OK done

**Comment by lightningterror** [comment]
Also worth keeping in mind, it's a good idea to also check DX9 behavior when testing. While it's on the verge of purging after 1.6 release, we don't want it to be even more broken until then.

**Pull request merged**